routes/cube_routes.js

```diff
@@ -422,42 +424,31 @@ router.get('/overview/:id', async (req, res) => {
       return res.status(404).render('misc/404', {});
     }
 
-    const pids = new Set();
-    for (const card of cube.cards) {
-      card.details = carddb.cardFromId(card.cardID);
-      const allVersions = carddb.getIdsFromName(card.details.name) || [];
-      card.allDetails = allVersions.map((id) => carddb.cardFromId(id));
-      for (const details of card.allDetails) {
-        if (details.tcgplayer_id) {
-          pids.add(details.tcgplayer_id);
-        }
-      }
-    }
-
+    const cardDetailsQ = populateCardDetails([cube.cards], carddb, prices, null, undefined, true);
     const userQ = User.findById(cube.owner);
     const blogsQ = Blog.find({
       cube: cube._id,
     }).sort('date');
-    const priceDictQ = GetPrices([...pids]);
-    const [user, blogs, priceDict] = await Promise.all([userQ, blogsQ, priceDictQ]);
+    const [user, blogs, cardDetails] = await Promise.all([userQ, blogsQ, cardDetailsQ]);
```

Comment:
cardDetails is not used anywhere in this route from what I can see. If populateCardDetails is modifying the collection that's passed in, that's an anti-pattern